lib/appsignal/rack.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Appsignal
+  # @api private
+  module Rack
+    # Alias constants that have moved with a warning message that points to the
+    # place to update the reference.
+    def self.const_missing(name)
+      case name
+      when :GenericInstrumentation
+        require "appsignal/rack/generic_instrumentation"
+
+        callers = caller
+        Appsignal::Utils::StdoutAndLoggerMessage.warning \
+          "The constant Appsignal::Rack::GenericInstrumentation has been deprecated. " \
+            "Please update the constant name to " \
+            "Appsignal::Rack::InstrumentationMiddleware in the following file to " \
+            "remove this message.\n#{callers.first}"
```

Comment:
This says to update the constant name to `Appsignal::Rack::InstrumentationMiddleware`, but the new middleware is to be used differently, right? Perhaps we should instead tell customers to read our Rake documentation?

Reply:
The only difference in behavior is that this new middleware doesn't default the action name to "unknown" (and we want people to also install the EventHandler). I don't alias to the new one because I don't want to silently break things, and it should be a deliberate action to update it in case anyone relies on this behavior.

I will also link to the docs from here 👍